Progressive Web App feature + favicon improvements

timephy opened the pull request:
# Progressive Web App

Addition of the `static/manifest.json` allows the Mensa Website to be installed as a "Standalone App" on any smartphone ("Add to Home Screen"-Feature).

# Favicon

Favicon "source" is `svg/favicon.svg` which is used to generate (`bin/generate-favicon.sh`) the `favicon-xx.png`s.

timephy committed:
gitignore

timephy committed:
improved favicon and icon

timephy committed:
small svg changes

timephy committed:
modified/added links to favicons and manifest

timephy committed:
refactor favicon generation

timephy committed:
added Progressive Web App manifest

ackxolotl commented on static/manifest.json:
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "TUM Mensa",
```
```suggestion
  "name": "Mensa-Homometer",
```

I think "Mensa-Homometer" would be a better name. "TUM Mensa" doesn't really reflect what this app is about?

ackxolotl commented on static/manifest.json:
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "TUM Mensa",
+  "short_name": "TUM Mensa",
```
```suggestion
  "short_name": "Mensa-Homometer",
```

ackxolotl commented on homometer/templates/homometer/index.html:
```diff
@@ -9,9 +9,10 @@
-		<link rel="apple-touch-icon" sizes="180x180" href="{% static 'apple-touch-icon.png' %}">
-		<link rel="icon" type="image/png" sizes="32x32" href="{% static 'favicon-32x32.png' %}">
-		<link rel="icon" type="image/png" sizes="16x16" href="{% static 'favicon-16x16.png' %}">
+		<link rel="apple-touch-icon" sizes="180x180" href="{% static 'favicon-180.png' %}">
```
```suggestion
		<link rel="apple-touch-icon" sizes="180x180" href="{% static 'favicon-180.png' %}" />
```

My bad. Some tags had `/>` and some didn't. Can you unify this and add `/` to this line and the following two? :)

ackxolotl commented on static/manifest.json:
```diff
@@ -0,0 +1,16 @@
+  "short_name": "TUM Mensa",
+  "icons": [
+    {
+      "src": "/images/favicon-192.png",
```
This path is wrong: It should be `/static` and there is no favicon of size `192x192` (yet).

ackxolotl reviewed (changes requested):
Thanks for your PR! I have some minor change requests. One big thing I noticed is that, although I like the color of the new favicon, it seems to be less recognizable than the old one (especially at `16x16`). Could you use [the old favicon](https://commons.wikimedia.org/wiki/File:AB-Rast.svg) or a better recognizable one?

timephy committed:
Update static/manifest.json

Co-Authored-By: Simon Ellmann <simon.ellmann@tum.de>

timephy committed:
Resolve most change requests

timephy committed:
Resolved most change requests

timephy committed:
icon scalings + fix icon path + change to more recognizable icon

timephy committed:
short name because full name does not fix on mobile

timephy committed:
undo last commit (selected wrong file)

timephy committed:
Merge branch 'master' of https://github.com/missing-user/mensa

timephy committed:
short name because full name does not fix on mobile

timephy commented:
Hey, thanks for the quick response!

I fixed all of the above issues. Changed to old icon with new background.
Fixed image path and html-typo.
Thanks for linking the original favicon symbol - I couldn't find it.

One more thing: I changed the PWA `short_name` to simply "Mensa", because the full name does not fit on mobile (see image).
![image](https://user-images.githubusercontent.com/28571656/77937098-1fb72f00-72b4-11ea-8d6d-14373a433886.png)

Hope you find the changes satisfactory!

ackxolotl commented on mensa/settings.py:
```diff
@@ -27,7 +27,7 @@
 DEBUG = True
 
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS = ["*"]
```
```suggestion
ALLOWED_HOSTS = []
```

Is this change necessary? `ALLOWED_HOSTS` shouldn't matter in debug mode.

timephy replied:
No this is definitely not necessary, I didn't mean to commit/push this change... Therefore I reverted this change immediately. I wanted to access the website from my phone. Since in debug the server rejected the phone's (non-`localhost`) request I changed this line to make it work.

ackxolotl reviewed (commented):
Looks good!

timephy committed:
Update mensa/settings.py

Co-Authored-By: Simon Ellmann <simon.ellmann@tum.de>

ackxolotl commented:
I've updated https://mensa.liste.party/. Hope it works as expected!
